### DISCUSSION TITLE
Admin privileges for specific users

### DISCUSSION THREAD

**Discussion by svdvonde** [yu-i-i/overleaf-cep#5] [Q&A] [answered]
Hi,

I'm trying to assign admin privileges for specific users by just hardcoding them instead of using an LDAP group, but my configuration options don't seem to be applying as I'd expect. Are my expectations wrong? 

in `config/variables.env` I set the following properties:

```text
OVERLEAF_LDAP_ADMIN_SEARCH_BASE=ou=People,dc=vub,dc=ac,dc=be
OVERLEAF_LDAP_ADMIN_SEARCH_FILTER=(uid=someuser)
OVERLEAF_LDAP_UPDATE_ADMIN_ON_LOGIN=true
```

When I log in with my LDAP user account (which is not `someuser`) I seem to have admin privileges in Overleaf. When I put `OVERLEAF_LDAP_ADMIN_SEARCH_FILTER` in comments I am (expectedly) no longer an admin.

**Comment by yu-i-i** [comment, marked as answer]
This is counterintuitive but expected behavior. The user gains admin status if the search result is not empty, not when the user is explicitly included in the search results.

The reasoning is as follows: usually, admins are members of the 'admins' group. The search starts from the base `cn=admins,ou=group,dc=example,dc=com` and checks whether the user's `{dn}` is included in the group's members. The result of this search is not a list of users, but rather the 'admins' group (or all groups containing the user as a member, if the search base is the root), or it could be empty. This is how the `passport-ldapauth` module works.

So, if you use the `uid` of an existing user as a search filter, the search result will always be non-empty, and every user will be granted admin rights! I should have written this in bold. My mistake for not doing so.

The example with the filter `'(&(gidNumber=1234)(uid={{username}}))'` shows how to address this issue. In your case, the filter should be: `'(&(uid=someuser)(uid={{username}}))'`.

If the user who has logged in is 'someuser', the search filter becomes `'(&(uid=someuser)(uid=someuser))'`, and the search result is not empty — it is 'someuser'. Otherwise, the filter will be `'(&(uid=someuser)(uid=anotheruser))'`, and the search result will be empty.

**Reply by svdvonde** [reply]
Heh, that makes sense. Thank you for educating me!
